contracts/farm/LiquidityMining.sol

```diff
@@ -385,19 +418,36 @@ contract LiquidityMining is LiquidityMiningStorage {
 	 * @param _user the address of user to claim reward from (can be passed only by wrapper contract)
 	 */
 	function claimReward(address _poolToken, address _user) external {
-		require(poolIdList[_poolToken] != 0, "Pool token not found");
 		address userAddress = _getUserAddress(_user);
 
 		uint256 poolId = _getPoolId(_poolToken);
-		PoolInfo storage pool = poolInfoList[poolId];
-		UserInfo storage user = userInfoMap[poolId][userAddress];
+		_claimReward(poolId, userAddress, true);
+	}
 
-		_updatePool(poolId);
+	function _claimReward(uint256 _poolId, address _userAddress, bool _isStakingTokens) internal {
+		PoolInfo storage pool = poolInfoList[_poolId];
+		UserInfo storage user = userInfoMap[_poolId][_userAddress];
+
+		_updatePool(_poolId);
 		_updateReward(pool, user);
-		_transferReward(user, userAddress, true);
+		_transferReward(address(pool.poolToken), user, _userAddress, _isStakingTokens, true);
 		_updateRewardDebt(pool, user);
 	}
 
+	/**
+	 * @notice transfers reward tokens from all pools
+	 * @param _user the address of user to claim reward from (can be passed only by wrapper contract)
+	 */
+	function claimRewardFromAllPools(address _user) external {
+		address userAddress = _getUserAddress(_user);
+
+		for (uint256 i = 0; i < poolInfoList.length; i++) {
```

Comment:
one can declare a local variable for `poolInfoList.length` to reduce gas consumption. Current code reads  `length` from storage on each iteration

Reply:
done